presto-main/src/test/java/com/facebook/presto/operator/TestUncheckedByteArrays.java

```diff
@@ -0,0 +1,142 @@
+                UncheckedByteArrays::getByteUnchecked);
+
+        testRandom((byte) 0, ARRAY_BYTE_INDEX_SCALE, UncheckedByteArrays::setByteUnchecked, UncheckedByteArrays::getByteUnchecked);
+    }
+
+    @Test
+    public static void testSetShort()
+    {
+        testSequential(
+                IntStream.range(0, POSITIONS_PER_PAGE).mapToObj(i -> (short) RANDOM.nextInt(256)).collect(Collectors.toList()),
+                ARRAY_SHORT_INDEX_SCALE,
+                UncheckedByteArrays::setShortUnchecked,
+                UncheckedByteArrays::getShortUnchecked);
+
+        testRandom((short) 0, ARRAY_SHORT_INDEX_SCALE, UncheckedByteArrays::setShortUnchecked, UncheckedByteArrays::getShortUnchecked);
+    }
+
+    @Test
+    public static void testSetInt()
+    {
+        testSequential(
+                IntStream.range(0, POSITIONS_PER_PAGE).boxed().collect(Collectors.toList()),
+                ARRAY_INT_INDEX_SCALE,
+                UncheckedByteArrays::setIntUnchecked,
+                UncheckedByteArrays::getIntUnchecked);
+
+        testRandom(5, ARRAY_INT_INDEX_SCALE, UncheckedByteArrays::setIntUnchecked, UncheckedByteArrays::getIntUnchecked);
+    }
+
+    @Test
+    public static void testSetLong()
+    {
+        testSequential(
+                LongStream.range(0, POSITIONS_PER_PAGE).boxed().collect(Collectors.toList()),
+                ARRAY_LONG_INDEX_SCALE,
+                UncheckedByteArrays::setLongUnchecked,
+                UncheckedByteArrays::getLongUnchecked);
+
+        testRandom(0L, ARRAY_LONG_INDEX_SCALE, UncheckedByteArrays::setLongUnchecked, UncheckedByteArrays::getLongUnchecked);
+    }
+
+    private static <T> void testSequential(List<T> source, int elementSize, TriFunction<byte[], Integer, T> writeFunction, BiFunction<byte[], Integer, T> readFunction)
+    {
+        byte[] destination = new byte[source.size() * elementSize];
+
+        int index = 0;
+        for (T value : source) {
+            index = writeFunction.apply(destination, index, value);
+        }
+
+        assertEquals(index, POSITIONS_PER_PAGE * elementSize);
+        assertCopied(source, destination, elementSize, readFunction);
+    }
+
+    private static <T> void testRandom(T sourceValue, int elementSize, TriFunction<byte[], Integer, T> writeFunction, BiFunction<byte[], Integer, T> readFunction)
+    {
+        byte[] destination = new byte[POSITIONS_PER_PAGE * elementSize];
+
+        int index = 0;
+        for (int i = 0; i < POSITIONS_PER_PAGE; i++) {
+            if (i % 2 == 0) {
+                index = writeFunction.apply(destination, index, sourceValue);
+            }
+        }
+
+        assertEquals(index, POSITIONS_PER_PAGE / 2 * elementSize);
+        assertCopied(
+                nCopies(POSITIONS_PER_PAGE / 2, sourceValue),
+                Arrays.copyOf(destination, POSITIONS_PER_PAGE * elementSize / 2),
+                elementSize,
+                readFunction);
+        assertCopied(
```

Comment:
this is a duplicate of L118; is there any particular reason to make this call twice?

Reply:
It should be only once.